node/src/config.rs

```diff
@@ -0,0 +1,326 @@
+use colored::Colorize;
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use crate::{
+    eth::backend::db::{Db, ForkedDatabase},
+    fork::ClientFork,
+    mem,
+    revm::db::CacheDB,
+};
+use ethers::{
+    core::k256::ecdsa::SigningKey,
+    prelude::{Address, Wallet, U256},
+    providers::{Middleware, Provider},
+    signers::{coins_bip39::English, MnemonicBuilder, Signer},
+    utils::{format_ether, hex, WEI_IN_ETHER},
+};
+use foundry_evm::{
+    executor::fork::{BlockchainDb, BlockchainDbMeta, SharedBackend},
+    revm,
+    revm::{BlockEnv, CfgEnv, TxEnv},
+};
+use parking_lot::RwLock;
+
+pub const NODE_PORT: u16 = 8545;
+
+/// Configurations of the EVM node
+#[derive(Debug, Clone)]
+pub struct NodeConfig {
+    /// Chain ID of the EVM chain
+    pub chain_id: u64,
+    /// Default gas limit for all txs
+    pub gas_limit: U256,
+    /// Default gas price for all txs
+    pub gas_price: U256,
+    /// Signer accounts that will be initialised with `genesis_balance` in the genesis block
+    pub genesis_accounts: Vec<Wallet<SigningKey>>,
+    /// Native token balance of every genesis account in the genesis block
+    pub genesis_balance: U256,
+    /// Signer accounts that can sign messages/transactions from the EVM node
+    pub accounts: HashMap<Address, Wallet<SigningKey>>,
+    /// Configured block time for the EVM chain. Use `None` to mine a new block for every tx
+    pub automine: Option<Duration>,
+    /// port to use for the server
+    pub port: u16,
+    /// maximumg number of transactions in a block
+    pub max_transactions: usize,
```

Comment:
How does the `automine` option affect this? I'd have guessed `max_transactions` defaults to 1, and a block is auto-mined only once `max_transactions` txs have been sent. But seems this defaults to 1000, so that guess seems incorrect

Reply:
this is a very primitive settings, 
due to the way we mine the number of tx in a new block will usually be exactly 1, because we mine as soon as a new tx arrives, but this option limits how many of the available tx the miner can put in a new block.